pkg/build/frontend.go

```diff
@@ -188,6 +195,8 @@ func resolveStates(ctx context.Context, bopts *BOpts, platform ocispecs.Platform
 			}
 			name := strings.TrimSuffix(dref.FamiliarString(named), ":latest")
 
+			// This is the key format in which the frontend expects the image digest to be passed
+			// to the builder
```

Comment:
Maybe add "qualified reference is used here to match the key that buildkit will search for". What do you think?